### PR TITLE
Gm outputs safe point

### PR DESCRIPTION
**Pull request by JaredHryszko**
Describe Changes
---
Create a new method to save_gm_results. mutli_area outputs zone IDs as index/columns in the value distribution and names of areas in parameter suffixes.

Updates to test_single_area.py to make it work with the additional input arguments (output_path, mostly).

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Is the documentation building correctly with a clean version number?
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `setup.cfg`

### CODE REVIEW

File: src/caf/distribute/gravity_model/multi_area.py [hunk 1]
```diff
@@ -375,7 +377,7 @@ class MGMCostDistribution:
 
     name: str
     cost_distribution: cost_utils.CostDistribution
-    zones: np.ndarray
+    zones: pd.DataFrame #np.ndarray
```
Comment (isaac-tfn) [inline src/caf/distribute/gravity_model/multi_area.py]: I think if this is only the index, which is from 0->n-1 and is used for indexing numpy arrays, and values which are the actual zone numbers, then it is clearer if it is a Series rather than DataFrame

**Review by isaac-tfn — COMMENTED** [review]
Just one comment in code, and other than that I think it would be good to have row and column targets input as Series instead of arrays. Other than those, just need some tidying up (e.g. remove commented out code) and then good to go

File: src/caf/distribute/gravity_model/multi_area.py [hunk 2]
```diff
@@ -736,6 +771,11 @@ def calibrate(
             )
 
             results[dist.name] = result_i
+            # save results
+            save_path = os.path.join(output_path, dist.name)
+            # create each subfolder
+            os.makedirs(save_path)
+            result_i.save_gm_results(save_path=save_path)
         return results
```
Comment (isaac-tfn) [inline src/caf/distribute/gravity_model/multi_area.py:776]: Sorry one more comment, could the overall matrix also be saved to the output folder? So at this point self.achieved_distribution

Comment (JaredHryszko) [reply src/caf/distribute/gravity_model/multi_area.py:776]: Added a new method in core.py, however it is static. I couldn't quite get the right information out of the results class to feed into creating the matrix and saving it. A different approach is to save it directly from within the calibrate() or run() methods from the MultiAreaGravityModelCalibrator class

File: src/caf/distribute/gravity_model/multi_area.py [hunk 3]
```diff
@@ -179,7 +179,7 @@ class MultiCostDistribution:
     @classmethod
     def from_pandas(
         cls,
-        ordered_zones: pd.Series,
+        # ordered_zones: pd.Series,
```
Comment (isaac-tfn) [inline src/caf/distribute/gravity_model/multi_area.py]: Could this be removed completely if not needed?

File: src/caf/distribute/gravity_model/multi_area.py [hunk 4]
```diff
@@ -538,17 +537,23 @@ def __init__(
...
-        self.row_targets = row_targets
-        self.col_targets = col_targets
+        self.row_targets = row_targets.to_numpy()
+        self.col_targets = col_targets.to_numpy()
+        # check for mismatching dimensions
         if len(row_targets) != cost_matrix.shape[0]:
             raise IndexError("row_targets doesn't match cost_matrix")
         if len(col_targets) != cost_matrix.shape[1]:
             raise IndexError("col_targets doesn't match cost_matrix")
+        # check for mismatching zone orders
```
Comment (isaac-tfn) [inline src/caf/distribute/gravity_model/multi_area.py]: Could these checks a.) be before row and col targets are assigned, and b.) try to do some reordering in the case that e.g. rows have the same zones as costs but in a different order. At the moment these checks will fail in that instance, but it might be better to raise a warning and reorder if possible.

**Review by isaac-tfn — COMMENTED** [review]
I think I managed to view an old version of this and comment based on that. The comment about removing a commented out line ignore if you still see it. Only one small comment on the rows/targets/costs validation, but otherwise looks good!

File: src/caf/distribute/gravity_model/multi_area.py (hunk 4 above)
Comment (JaredHryszko) [reply src/caf/distribute/gravity_model/multi_area.py]: Good point, amended the code to your comment, ran black/pylint then committed and pushed
